scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,314 @@
+        n : int
+            The number of trials (copied from `binomtest` input).
+        alternative : str
+            Indicates the alternative hypothesis specified in the input
+            to `binomtest`.  It will be one of ``'two-sided'``, ``'greater'``,
+            or ``'less'``.
+        pvalue : float
+            The p-value of the hypothesis test.
+        proportion_estimate : float
+            The estimate of the proportion of successes.
+
+        The object has the following methods:
+
+        proportion_ci(confidence_level=0.95, method='exact') :
+            Compute the confidence interval for ``proportion_estimate``.
+
+    References
+    ----------
+    .. [1] Binomial test, https://en.wikipedia.org/wiki/Binomial_test
+    .. [2] Jerrold H. Zar, Biostatistical Analysis (fifth edition),
+           Prentice Hall, Upper Saddle River, New Jersey USA (2010)
+
+    Examples
+    --------
+    >>> from scipy.stats import binomtest
+
+    A car manufacturer claims that no more than 10% of their cars are unsafe.
+    15 cars are inspected for safety, 3 were found to be unsafe. Test the
+    manufacturer's claim:
+
+    >>> result = binomtest(3, n=15, p=0.1, alternative='greater')
+    >>> result.pvalue
+    0.18406106910639114
+
+    The null hypothesis cannot be rejected at the 5% level of significance
+    because the returned p-value is greater than the critical value of 5%.
+
+    The estimated proportion is simply ``3/15``:
+
+    >>> result.proportion_estimate
+    0.2
+
+    We can use the `proportion_ci()` method of the result to compute the
+    confidence interval of the estimate:
+
+    >>> result.proportion_ci(confidence_level=0.95)
+    ConfidenceInterval(low=0.056846867590246826, high=1.0)
+
+    """
+    k = _validate_nonneg_int(k, 'k')
+    n = _validate_nonneg_int(n, 'n')
+    if k > n:
+        raise ValueError('k must not be greater than n.')
+
+    if not (0 <= p <= 1):
+        raise ValueError("p must be in range [0,1]")
+
+    if alternative not in ('two-sided', 'less', 'greater'):
+        raise ValueError("alternative not recognized; \n"
+                         "must be 'two-sided', 'less' or 'greater'")
+
+    if alternative == 'less':
+        pval = binom.cdf(k, n, p)
+    elif alternative == 'greater':
+        pval = binom.sf(k-1, n, p)
+    else:
+        # alternative is 'two-sided'
+        d = binom.pmf(k, n, p)
+        rerr = 1 + 1e-7
+        if k == p * n:
+            # special case as shortcut, would also be handled by `else` below
+            pval = 1.
+        elif k < p * n:
+            i = np.arange(np.ceil(p * n), n+1)
+            y = np.sum(binom.pmf(i, n, p) <= d*rerr, axis=0)
+            pval = binom.cdf(k, n, p) + binom.sf(n - y, n, p)
+        else:
+            i = np.arange(np.floor(p*n) + 1)
+            y = np.sum(binom.pmf(i, n, p) <= d*rerr, axis=0)
+            pval = binom.cdf(y-1, n, p) + binom.sf(k-1, n, p)
```

Comment:
For any future reviewer, this is essentially equivalent to
```
ks = np.arange(n+1)
dist = binom(n=n, p=p)
pmf = dist.pmf(ks)   # the whole probability mass function
pmf_k = dist.pmf(k)  # the probability mass of the observed number of successes
j = pmf <= pmf_k*rerr
pval = pmf[j].sum()
```
but presumably faster.